### ISSUE TITLE
--cert-restart-on-secret-refresh not possible when not using different flag parsing framework

### ISSUE THREAD

**Issue by stijndehaes** [open-policy-agent/cert-controller#19]
I use a different flag framework to parse flags, and would like this to be a configuration option you pass to the rotator when starting up. However this might be a breaking change for other users. How do you feel about changing this?

**Comment by adrianludwin** [comment]
What happens if we give the _option_ to set this option when you're starting up the rotator? That way, if you never call `flag.Parse()`, you get to pick what you want the value to be. But if you _do_ call `flag.Parse()` and _don't_ pass in the override, then we'll maintain the current behaviour. Sgty?

**Comment by stijndehaes** [comment]
@adrianludwin that would be fine for me. 

So we add the option to the `CertRotator` struct and in the `AddRotator` call we do something like:

```
func AddRotator(mgr manager.Manager, cr *CertRotator) error {
  restartOnSecretRefresh = restartOnSecretRefresh || cr.restartOnSecretRefresh 
  ...
}
```

**Comment by adrianludwin** [comment]
Yup that lgtm (looks good to me)! Note that I'm not an owner of this project but I suspect that @maxsmythe , @ritazh etc would be happy with this as well.

**Comment by maxsmythe** [comment]
Not sure I like the idea of mixing the two. Users who do use the standard flag parser may accidentally also set the config option, which would negate the config.

How do people feel about...

1. Starting tagging commits with semver so that we can track non-backwards-compatible changes
2. Removing the flag and implementing a config option, incrementing this semver

Projects using this would then be able to just implement their flag individually should they want to keep it.

I'm not sure how many consumers of this project there are, but hopefully the semver stuff will give them enough warning to avoid surprise.

@ritazh @shomron thoughts?

**Comment by adrianludwin** [comment]
+100 to semver. I kind of liked "batteries included" but I guess that does
make it a bit worse as a library, so I could live with that!

**Comment by maxsmythe** [comment]
Yeah, working with some of these other libraries that use flags... the batteries included approach works great until you need to change the shape of the batteries :p

**Comment by adrianludwin** [comment]
sgtm

**Comment by stijndehaes** [comment]
I would be all for doing semver and making this a breaking change in a major release or something similar :)

**Comment by ritazh** [comment]
+1 on semver and start cutting releases for this project. It would make introducing breaking changes much easier.

**Comment by adrianludwin** [comment]
@ritazh , @maxsmythe - what do we need to do to make this happen? I'm not an admin on this repo so I can't create a release myself. Can we just call what we currently have "v0.1.0" and then start working towards v0.2.0?

**Comment by ritazh** [comment]
Since we don't have CI to automate releases yet until https://github.com/open-policy-agent/cert-controller/issues/14 is done, I could manually create/push a tag with the latest commit in master. WDYT @maxsmythe?

**Comment by maxsmythe** [comment]
SGTM

**Comment by ritazh** [comment]
Sorry for the delay. 
First release 🎉 
https://github.com/open-policy-agent/cert-controller/releases/tag/v0.1.0

**Comment by stijndehaes** [comment]
I made a PR for making this option part of the CertRotator struct: https://github.com/open-policy-agent/cert-controller/pull/23

**Comment by maxsmythe** [comment]
Thanks!
